spec/features/search_page_spec.rb

```diff
@@ -63,4 +63,80 @@
 
   end
 
+  scenario 'Match location query against available locations', :js => true do
+
+    first_dataset = DatasetBuilder.new
+                      .with_title('Wellington Dataset')
+                      .with_location('Wellington')
+                      .build
+
+    second_dataset = DatasetBuilder.new
+                       .with_title('Welding Dataset')
+                       .with_location('Welding')
+                       .build
+
+    index([first_dataset, second_dataset])
+
+    visit('/search')
+
+    assert_data_set_length_is(2)
+
+    page.driver.execute_script %Q{ $('#location').focus().typeahead('val', 'We') }
+    page.driver.execute_script %Q{ $('#location').focus().typeahead('open') }
+
+    expect(page).to have_selector('.tt-selectable', count: 2)
+    expect(page).to have_css('.tt-selectable', text: 'Wellington')
+    expect(page).to have_css('.tt-selectable', text: 'Welding')
+
+    page.driver.execute_script %Q{ $('#location').focus().typeahead('val', 'Wellington') }
```

Comment:
This is tied to our implementation, so might prove brittle. But I am guessing you did this because just filling in the field didn't work?

Reply:
I tried for ages to try and get Capybara to do this without executing javascript but had no luck. I was reading that apparently trying to get Capybara to play nicely with third party javascript can be a bit hit and miss - it's often the driver that the unknown quantity. Then again, I am a Capybara noob and might not be doing things correctly. Would you be up for having a go?